src/cmdstan/stansummary.cpp

```diff
@@ -36,131 +36,68 @@ Example:  stansummary model_chain_1.csv model_chain_2.csv
   -s, --sig_figs [n]          Significant figures reported. Default is 2.
                               Must be an integer from (1, 18), inclusive.
 )";
-  if (argc < 2) {  // pre-empt boost::program_options
+  if (argc < 2) {
     std::cout << usage << std::endl;
     return -1;
   }
 
   // Command-line arguments
-  int sig_figs;
+  int sig_figs = 2;
   int autocorr_idx;
   std::string csv_filename;
-  std::string percentiles_spec;
+  std::string percentiles_spec = "5,50,95";
   std::vector<std::string> filenames;
-  boost::program_options::options_description desc("Allowed options");
-  desc.add_options()("help,h", "Produce help message")(
-      "sig_figs,s",
-      boost::program_options::value<int>(&sig_figs)->default_value(2),
-      "Significant figures, default 2.")(
-      "autocorr,a", boost::program_options::value<int>(&autocorr_idx),
-      "Display the chain autocorrelation.")(
-      "csv_filename,c",
-      boost::program_options::value<std::string>(&csv_filename),
-      "Write statistics to a csv.")(
-      "percentiles,p",
-      boost::program_options::value<std::string>(&percentiles_spec)
-          ->default_value("5,50,95"),
-      "Percentiles to report.")(
-      "input_files,i",
-      boost::program_options::value<std::vector<std::string> >(&filenames),
-      "Sampler csv files. ");
-  boost::program_options::positional_options_description p;
-  p.add("input_files", -1);
 
-  // Parse, validate command-line
-  boost::program_options::variables_map vm;
+  CLI::App app{"Allowed options"};
+  app.add_option("--sig_figs,-s", sig_figs, "Significant figures, default 2.",
+                 true)
+      ->check(CLI::Range(1, 18));
+  app.add_option("--autocorr,-a", autocorr_idx,
+                 "Display the chain autocorrelation.", true)
+      ->check(CLI::PositiveNumber);
+  app.add_option("--csv_filename,-c", csv_filename,
+                 "Write statistics to a csv.", true)
+      ->check(CLI::NonexistentPath);
+  app.add_option("--percentiles,-p", percentiles_spec, "Percentiles to report.",
+                 true);
+  app.add_option("input_files", filenames, "Sampler csv files.", true)
+      ->required()
+      ->check(CLI::ExistingFile);
```

Comment:
This checks the input files are readable?

Reply:
checks that they exist.
need to add checks for readibility, similar to check for writable output csv file.

Reply:
done

Reply:
Totally optional, but when I get the error for readability I don't get the friendly `--help` prompt:

```
bbales2@tadpole:~/cmdstan-develop$ bin/stansummary output4.csv
Cannot read input csv file: output4.csv.
```

Compare this to:

```
bbales2@tadpole:~/cmdstan-develop$ bin/stansummary output12321.csv
input_files: File does not exist: output12321.csv
Run with --help for more information.
```

Reply:
Also I don't get the --help prompt with the percentiles option:

```
bbales2@tadpole:~/cmdstan-develop$ bin/stansummary -p -5 output4.csv
Option --percentiles -5: values must be in range (1,99), inclusive, and strictly increasing.
```

Very optional to add this.

Reply:
can't figure out how to plug in custom help message via CLI11 - spent several hours looking.
will give it another go.

Reply:
Eh if it's not easy let's not worry about it.

Reply:
doable but not easy.  worth doing as part of next PR.